File: golang/leetcode/twosum.go
package leetcode

func TwoSum(nums []int, target int) []int {

	var output []int

	for i, v := range nums {
		for subI, subV := range nums {
			if i != subI && v+subV == target {
				output = append(output, i)
				output = append(output, subI)
				break
			}
		}
	}

	return RemoveDuplicates(output)
}

func RemoveDuplicates(slice []int) []int {
	found := make(map[int]bool)
	var result []int

	for _, value := range slice {
		if _, exists := found[value]; !exists {
			found[value] = true
			result = append(result, value)
		}
	}

	return result
}

File: leetcode_google/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        m = {}
        n = len(nums)
        for i in range(n):
            complement = target - nums[i]
            if complement in m:
                return [m[complement], i]
            m[nums[i]] = i
        return []

nums = [2,7,11,15]
target = 9
result = Solution().twoSum(nums, target)
print(result)
